sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumns.scala

```diff
@@ -522,6 +523,20 @@ case class ResolveDefaultColumns(
       case _ => return None
     }
     val lookup: LogicalPlan = try {
+      // Check if the target table has "ACCEPT_ANY_SCHEMA" capabilities and if so,
+      // don't do anything.
+      analyzer.currentCatalog match {
```

Comment:
In such a case, we need to find a way to make sure table relations are resolved first and the analyzer doesn't failed on unresolved default expressions.

Reply:
SG, done.

This match should be OK since it looks like in the [last PR](https://github.com/apache/spark/pull/36445) we updated the `resolveMergeExprOrFail` method in the `Analyzer` to exclude unresolved default references specifically so that it would let the `ResolveDefaultColumns` analyzer rule handle them (which should either replace them all or else return an error):

<img width="937" alt="image" src="https://user-images.githubusercontent.com/99207096/169120472-a8cb08d9-3e79-44b9-893c-709c39af5d4f.png">